src/gui/options_banlist.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  Alexandr Milovantsev <dzmat@yandex.ru>
+ * Copyright (C) 2006  Christophe Dumez <chris@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "options_banlist.h"
+#include "ui_options_banlist.h"
+
+#include <QMessageBox>
+
+#include "base/bittorrent/session.h"
+
+options_banlist::options_banlist(QWidget *parent)
+    : QDialog(parent)
+    , m_ui(new Ui::options_banlist)
+    , modified(false)
+{
+    m_ui->setupUi(this);
+    m_ui->bannedIPList->addItems(BitTorrent::Session::instance()->getBannedIPList());
+}
+
+options_banlist::~options_banlist()
+{
+    delete m_ui;
+}
+
+void options_banlist::on_buttonBox_accepted()
+{
+    if (modified) {
+        // save to session
+        QStringList IPList;
+        for (int i = 0; i < m_ui->bannedIPList->count(); ++i)
+            IPList << m_ui->bannedIPList->item(i)->text();
+        BitTorrent::Session::instance()->setBannedIPList(IPList);
+    }
+    accept();
+}
+
+void options_banlist::on_pb_BanIP_clicked()
+{
+    QString s = m_ui->ed_IP->text();
+    if (s.isEmpty()) {
+        QMessageBox::warning(this, tr("Warning"), tr("At first enter IP address for bannig, please."));
```

Comment:
The message is even more cryptic now, as far as I'm concerned. Maybe it would be better to disable the button when the text box is empty?